identity-service/src/routes/authentication.js

```diff
@@ -30,7 +31,19 @@ module.exports = function (app) {
 
     if (body && body.iv && body.cipherText && body.lookupKey) {
       try {
-        await models.Authentication.create({ iv: body.iv, cipherText: body.cipherText, lookupKey: body.lookupKey })
+        await sequelize.transaction(function (t) {
+          return models.Authentication.create({
+            iv: body.iv,
+            cipherText: body.cipherText,
+            lookupKey: body.lookupKey
+          }, { transaction: t })
+            .then(function (auth) {
+              const oldLookupKey = body.oldLookupKey
+              if (oldLookupKey) {
+                return models.Authentication.destroy({ where: { lookupKey: oldLookupKey } }, { transaction: t })
```

Comment:
in the event the old and new lookup keys are the same, wouldn't this set the row as deleted?

Reply:
I tested that case, and since they're in the same transaction the deletion no-ops and the sign up succeeds. But this does have me think about the "edit" scenario - though wouldn't that have a new lookup key?

Reply:
it seems bad to rely on some implicit behavior of this tx block for the creation & deletion. it seems like the DMS rather than the ORM that has this constraint. could we just add our own equality check and not destroy when they're the same?

Reply:
Agreed.